### PR TITLE
shutdown cluster by default

### LINKED ISSUE #611: In-place restores should shutdown the cluster by default
**Is your feature request related to a problem? Please describe.**
In-place restores are done via rolling restart by default. If the cluster remains available and continues accepting writes while the restore is in process, we cannot guarantee that the cluster will restored to the backup state. 

Furthermore, there is a possibility that writes could be lost. During the restore process Medusa deletes the contents of the data subdirectory of the table being restored. This could easily lead to data loss.

**Describe the solution you'd like**
Shut down the entire cluster by default to be proactive in preventing data loss or inconsistent state during a restore.

It is worth noting that shutting down the cluster won't completely prevent these problems. As nodes come back online, clients could still perform queries before the restore is complete. Completely preventing client access is beyond the scope of this issue and might be something we can better address when we do service mesh integration.

**Describe alternatives you've considered**

**Additional context**
Restores are currently are really, really slow. Part of that is due to #570 and #571. 

The other reason is that the restore operation is done serially one Cassandra pod at a time. The restore operation is performed by the `medusa-restore` init container. The bulk of the time of the restore operation will be from downloading the backup files. Doing it serially one node at a time will make it painfully slow and could result in result in long periods of downtime for the cluster. 

The reason it is done serially is because the restore happens by way of a StatefulSet update. The StatefulSet controller does a rolling update of pods, one at a time. I think we can work around this relatively easily. The StatefulSet's `.spec.podManagementPolicy` property controls how pods are created. The default value is `OrderedReady` which means one pod at a time. cass-operator uses `Parallel`. If we simply delete the StatefulSet(s) and let them get recreated, the `medusa-restore` init containers will run in parallel. Note that deleting a Statefulset does not cascade to the PersistentVolumeClaims and PersistentVolumes that it uses.

### PR DESCRIPTION
**Pull request by jsanda**
**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #611
**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: charts/restore/values.yaml [hunk 1]
```diff
@@ -11,13 +11,20 @@ cassandraDatacenter:
...
-# been restored.
-shutdown: false
+# backup includes schema changes like dropping a table, then must be set to
+# true; otherwise, the changes will be lost via gossip from nodes that have
+# not yet been restored. It is recommended in general to shutdown the cluster
+# prior to a restore to avoid data inconsistencies and/or data loss that could
+# happen with clients writing to the cluster while the restore operation is in
+# progress.
+#
+# When set the cluster is shutdown the restore operations happen in parallel
```
Comment (emerkle826) [inline charts/restore/values.yaml:26]: nit: add the word `and`

_suggested change_:
```
When set the cluster is shutdown and the restore operations happen in parallel
```

**Review by emerkle826 — APPROVED** [review]
small nitpick if you want to address it.

File: charts/restore/values.yaml (hunk 1 above)
Comment (jsanda) [reply charts/restore/values.yaml:26]: Yup, I want to address it :)

**Comment by burmanm** [comment]
Is this missing a file in the commit? There's no changes to the backup.yaml template

**Comment by jsanda** [comment]
> Is this missing a file in the commit? There's no changes to the backup.yaml template

It's there now. The only change is in charts/restore/values.yaml.

**Pull request merged**